tests/cuistl/test_cujac.cpp

```diff
@@ -0,0 +1,271 @@
+    Opm::cuistl::CuVector<T> d_invDiag(blocksize*blocksize*N);
+
+    Opm::cuistl::detail::invertDiagonalAndFlatten(m.getNonZeroValues().data(), m.getRowIndices().data(), m.getColumnIndices().data(), N, blocksize, d_invDiag.data());
+
+    std::vector<T> expected_inv_diag{2.0,-2.0,-1.0/2.0,1.0,-1.0,0.0,0.0,-1.0};
+    std::vector<T> computed_inv_diag = d_invDiag.asStdVector();
+
+    BOOST_REQUIRE_EQUAL(expected_inv_diag.size(), computed_inv_diag.size());
+    for (size_t i = 0; i < expected_inv_diag.size(); i++){
+        BOOST_CHECK_CLOSE(expected_inv_diag[i], computed_inv_diag[i], 1e-7);
+    }
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(ElementWiseMultiplicationOf3By3BlockVectorAndVectorVector, T, NumericTypes)
+{
+    /*
+        Example in the test for multiplying by element a blockvector with a vector of vectors
+        | |1 2 3| |   | |3| |   | |10| |
+        | |5 2 3| | X | |2| | = | |22| |
+        | |2 1 1| |   | |1| |   | |10| |
+    */
+
+    const size_t blocksize = 3;
+    const size_t N = 1;
+    
+    std::vector<T> h_blockVector({1.0,2.0,3.0,5.0,2.0,3.0,2.0,1.0,2.0});
+    std::vector<T> h_vecVector({3.0,2.0,1.0});
+    Opm::cuistl::CuVector<T> d_blockVector(h_blockVector);
+    Opm::cuistl::CuVector<T> d_vecVector(h_vecVector); 
+
+    Opm::cuistl::detail::blockVectorMultiplicationAtAllIndices(d_blockVector.data(), N, blocksize, d_vecVector.data());
+
+    std::vector<T> expected_vec{10.0,22.0,10.0};
+    std::vector<T> computed_vec = d_vecVector.asStdVector();
+
+    BOOST_REQUIRE_EQUAL(expected_vec.size(), computed_vec.size());
+    for (size_t i = 0; i < expected_vec.size(); i++){
+        BOOST_CHECK_CLOSE(expected_vec[i], computed_vec[i], 1e-7);
+    }
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(ElementWiseMultiplicationOf2By2BlockVectorAndVectorVector, T, NumericTypes)
+{
+    /*
+        Example in the test for multiplying by element a blockvector with a vector of vectors
+        | |1 2| |   | |1| |   | | 7| |
+        | |3 4| | X | |3| | = | |15| |
+        |       |   |     |   |      |
+        | |4 3| |   | |2| |   | |20| |
+        | |2 1| |   | |4| |   | | 8| |
+    */
+
+    const size_t blocksize = 2;
+    const size_t N = 2;
+    
+    std::vector<T> h_blockVector({1.0,2.0,3.0,4.0,4.0,3.0,2.0,1.0});
+    std::vector<T> h_vecVector({1.0,3.0,2.0,4.0});
+    Opm::cuistl::CuVector<T> d_blockVector(h_blockVector);
+    Opm::cuistl::CuVector<T> d_vecVector(h_vecVector); 
+
+    Opm::cuistl::detail::blockVectorMultiplicationAtAllIndices(d_blockVector.data(), N, blocksize, d_vecVector.data());
+
+    std::vector<T> expected_vec{7.0,15.0,20.0,8.0};
+    std::vector<T> computed_vec = d_vecVector.asStdVector();
+
+    BOOST_REQUIRE_EQUAL(expected_vec.size(), computed_vec.size());
+    for (size_t i = 0; i < expected_vec.size(); i++){
+        BOOST_CHECK_CLOSE(expected_vec[i], computed_vec[i], 1e-7);
+    }
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(CUJACApplyIsEqualToDuneSeqJacApply, T, NumericTypes)
```

Comment:
I think this file should be split in three, keeping this test in the current file but moving the others to `test_vector_operations.cpp` and `test_cusparse_matrix_operations.cpp`. Making the filenames match makes it easier to locate the tests.

Reply:
agree